lib/events/api.go

```diff
@@ -489,6 +489,22 @@ type MultipartUploader interface {
 	// ListUploads lists uploads that have been initiated but not completed with
 	// earlier uploads returned first
 	ListUploads(ctx context.Context) ([]StreamUpload, error)
+	// GetUploadMetadata gets the upload metadata
+	GetUploadMetadata(sessionID session.ID) UploadMetadata
+}
+
+// UploadMetadata contains data about the session upload
+type UploadMetadata struct {
+	// URL is the url at which the session recording is located
+	// it is free-form and uploader-specific
+	URL string
+	// SessionID is the event session ID
+	SessionID session.ID
+}
+
+// UploadMetadataGetter gets the metadata for session upload
+type UploadMetadataGetter interface {
+	GetUploadMetadata(sid session.ID) *UploadMetadata
```

Comment:
nit: this can also return by value so it matches `MultipartUploader.GetUploadMetadata` - you can look at it as if `MultipartUploader` also implemented the same interface.